src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4059,6 +4110,30 @@ def create_functionapp(cmd, resource_group_name, name, storage_account, plan=Non
     return functionapp
 
 
+def _validate_cpu_momory_functionapp(cpu=None, memory=None):
+    # validate either both cup and momory are provided or none is provided. throw error otherwise
+    if cpu is None and memory is None:
+        return
+
+    if cpu is not None and memory is None:
+        raise ArgumentUsageError("The --memory argument is required with --cpu. Please provide both or none.")
+
+    if cpu is None and memory is not None:
+        raise ArgumentUsageError("The --cpu argument is required with --memory. Please provide both or none.")
+
+    # validate that memory is number and ends with Gi
+    if memory is not None and not memory.lower().endswith("gi"):
+        raise ValidationError("The --memory argument should end with Gi. Please provide a correct value. e.g. 4.0Gi.")
+
+    if memory is not None:
```

Comment:
Nit: I think by the time this logic runs, memory is guaranteed not to be `None` so there is no need to validate that anymore for these two if statements.